### ISSUE TITLE
pactsafe-angular-sdk giving compiler error when upgraded angular project from version 8 to 10.

### ISSUE THREAD

**Issue by RupaliBochare** [pactsafe/pactsafe-angular-sdk#10]
Hi @JJWill0ughby @eprugh,

I have AngularCli generated project initally version is 8 there i am using  "pactsafe-angular-sdk": "^0.1.2", and all working fine.

Now currently i am upgrading angular version to 10 and facing issue while compilation as below 

[INFO] chunk {} runtime.acf0dec4155e77772545.js (runtime) 1.45 kB [entry] [rendered]
[INFO] ERROR in node_modules/pactsafe-angular-sdk/index.d.ts:5:23 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).
[INFO] chunk {1} main.32a5bf83ddf75a6d3667.js (main) 128 bytes [initial] [rendered]
[INFO]
[INFO] chunk {2} polyfills.4b1f18e0602c4a1bab0a.js (polyfills) 130 bytes [initial] [rendered]
[INFO] 5     static forRoot(): ModuleWithProviders;
[INFO] chunk {3} polyfills-es5.b767e4712bb767052abe.js (polyfills-es5) 83 kB [initial] [rendered]

I also tried solution given in notes. i.e

If you are developing in an AngularCli generated project, you may run into trouble using the library. Make sure you set up a path mapping in /src/tsconfig.app.json of your consuming application (not the library) and that should solve any compilation problems you may have:

{
  "compilerOptions": {
    // ...
    // Note: these paths are relative to 'baseUrl' path.
    "paths": {
      "@angular/*": [
        "../node_modules/@angular/*"
      ]
    }
  }
}
still facing issue.
I am not much clear on what all paths need to mention here . cloud you please provide some inputs .

Compiler option in my projects

 "compilerOptions": {
    "outDir": "./out-tsc/app",
    "types": [],
    "paths": {
      "@angular/*": [
        "../node_modules/@angular/*"    
      ]
    }
  }

**Comment by RupaliBochare** [comment]
Any update on this issue.

**Comment by eprugh** [comment]
Hi there @RupaliBochare — apologies for not replying to this issue. We will have our team look into this and see if we can come back with some potential fixes that we can recommend. Thanks for your patience.

**Comment by eprugh** [comment]
Hi @RupaliBochare — can you take a look at the following URL? https://angular.io/guide/migration-module-with-providers

The fix may be as simple as editing the `index.ts` file to go from...

```export class PSModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: PSModule,
      providers: []
    };
  }
}```

to be...

```export class PSModule {
  static forRoot(): ModuleWithProviders<PSModule> {
    return {
      ngModule: PSModule,
      providers: []
    };
  }
}```

Can you please try that and see if you're able to get the SDK running?

Thanks!

**Comment by RupaliBochare** [comment]
@eprugh we already try this solution and sdk is working with this change but we cant change the index.ts file as we have include pactsafe as "pactsafe-angular-sdk": "^0.1.2" in package.json and we are not bundle node module in code base meaning we are installing package mention in package.json in every build.

When new build happens,  packages mention in package.json got installed fresh and its run time so we cant change the index.d.ts file on deployed envrioment. 

So ideally we expacting when we install pactsafe sdk package then index.ts file should have 
 static forRoot(): ModuleWithProviders<PSModule> {
    return {
      ngModule: PSModule,
      providers: []
    };
  }
}

**Comment by nutankhamkar** [comment]
We need updated version of pactsafe angular sdk which will be compatible with angular 10.....our angular version upgrade is blocked due to this

**Comment by eprugh** [comment]
@nutankhamkar — we are working on a fix for this and should have something live soon. We will update you as soon as we get that change pushed.

**Comment by mewelling** [comment]
@nutankhamkar would you be able to checkout the pull request i just created and let me know if that solution works for you?

**Comment by nutankhamkar** [comment]
Ok will checkout and let you know if it works..

**Comment by nutankhamkar** [comment]
Hi,
I have tried solution given by you in pull request, it's not working getting below error

[INFO] chunk {1} main.32a5bf83ddf75a6d3667.js (main) 128 bytes [initial] [rendered]ERROR in node_modules/pactsafe-angular-sdk/index.d.ts:5:53 - error TS1183: An implementation cannot be declared in ambient contexts.
[INFO] chunk {1} main.32a5bf83ddf75a6d3667.js (main) 128 bytes [initial] [rendered]ERROR in node_modules/pactsafe-angular-sdk/index.d.ts:5:53 - error TS1183: An implementation cannot be declared in ambient contexts.
[INFO]


please check

**Comment by nutankhamkar** [comment]
any update on this we are blocked

**Comment by brdly-smith** [comment]
@nutankhamkar there were some breaking changes to the library in version 9 of Angular with `ModuleWithProviders` becoming generic.

We have identified the current issue and are working on publishing an updated release candidate package compatible with Angular 10 for your testing.

**Comment by brdly-smith** [comment]
@nutankhamkar We have published a new package for this, please update your package.json to use:

`"@pactsafe/pactsafe-angular-sdk": "^2.0.0"`

If you run into further issues, please let us know in a new issue on this repo.

**Comment by nutankhamkar** [comment]
sure thanks for the reply will check and let you know

**Comment by nutankhamkar** [comment]
Hi,
I have installed updated package given by you, getting below errors:

ERROR in Unexpected value 'PSModule in C:/TMSWithSpringBoot/tenant-management-portal/tenant-ui/node_modules/@pactsafe/pactsafe-angular-sdk/index.d.ts' imported by the module 'RegisterModule in C:/TMSWithSpringBoot/tenant-management-portal/tenant-ui/src/app/register/register.module.ts'. Please add a @NgModule annotation.
[INFO] chunk {} runtime.acf0dec4155e77772545.js (runtime) 1.45 kB [entry] [rendered]Can't bind to 'signerIdSelector' since it isn't a known property of 'ps-click-wrap'.
[INFO] chunk {} runtime.acf0dec4155e77772545.js (runtime) 1.45 kB [entry] [rendered]Can't bind to 'signerIdSelector' since it isn't a known property of 'ps-click-wrap'.
[INFO] 1. If 'ps-click-wrap' is an Angular component and it has 'signerIdSelector' input, then verify that it is part of this module.
[INFO] 2. If 'ps-click-wrap' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
[INFO] 3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("-wrap accessId="eb356a5c-459a-4d99-b734-d19e15e6e64e" groupKey="trial-and-marketplace"
[INFO]
[INFO] chunk {1} main.32a5bf83ddf75a6d3667.js (main) 128 bytes [initial] [rendered]
[INFO]             [ERROR ->]signerIdSelector="{{id}}_email" (change)="termsAndCondition()"></ps-click-wrap>
[INFO]         <div>
[INFO]      ")
[INFO] 'ps-click-wrap' is not a known chunk {2} polyfills.4b1f18e0602c4a1bab0a.js (polyfills) 130 bytes [initial] [rendered]element:
[INFO] 1. If 'ps-click-wrap' is an Angular component, then verify that it is part of this module.
[INFO] 2. If 'ps-click-wrap' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("STER.COUNTRY' | translate }}" [settings]="countrySelectSettings">
[INFO]         </adapt-select>
[INFO] 'ps-click-wrap' is not a known chunk {2} polyfills.4b1f18e0602c4a1bab0a.js (polyfills) 130 bytes [initial] [rendered]
[INFO]         [ERROR ->]<ps-click-wrap accessId="eb356a5c-459a-4d99-b734-d19e15e6e64e" groupKey="trial-and-marketplace"
[INFO] chunk {3} polyfills-es5.b767e4712bb767052abe.js (polyfills-es5) 83 kB [initial] [rendered]
[INFO] chunk {4} styles.a273e9172c0607b9cd01.css (styles) 877 kB [initial] [rendered]
[INFO]     ")
[INFO]     ")Date: 2021-05-27T05:51:09.680Z - Hash: ca75a249010170b0c41c - Time: 56048ms

**Comment by nutankhamkar** [comment]
Hi,
I have received mail from 'Michael Welling', in that mail one demo app is given for which I don't have access, also I replied to that mail that I have requested access, but no reply.
Could you please give access as soon as possible?

Thanks,
Nutan

**Comment by nutankhamkar** [comment]
Hi Michael
Got the access for the App. I have tested Demo App given by you, it helped us to resolve the issue. Now we are unblocked.
Thanks a lot!!

Regards,
Nutan
